src/data/rat/defs.lean

```diff
@@ -465,10 +466,13 @@ instance : comm_ring ℚ :=
   mul_assoc        := rat.mul_assoc,
   left_distrib     := rat.mul_add,
   right_distrib    := rat.add_mul,
-  nat_cast         := λ n, rat.of_int n,
+  int_cast         := coe,
+  /- Important: We do not set `nat_cast := λ n, ((n : ℤ) : ℚ)` (even though it's defeq) as that
+  makes `int.cast_coe_nat` and `coe_coe` loop in `simp`. -/
+  nat_cast         := λ n, ⟨n, 1, nat.one_pos, nat.coprime_one_right _⟩,
```

Comment:
Does it still go wrong if you use
```suggestion
  nat_cast         := λ n, has_int_cast.int_cast n,
```
or similar? That would at least avoid repeating this term, though you might need

Reply:
It does work but it's slightly worse computationally given that it uses two coercions instead of one.

Reply:
Isn't the computational content identical?

Reply:
I guess it doesn't make much difference.